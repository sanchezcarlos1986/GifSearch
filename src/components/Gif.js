import React, {memo} from 'react';
import {Link} from 'wouter';
import PropTypes from 'prop-types';

/**
 * Represents a block with the gif's content.
 * @constructor
 * @param {string} id - Gif's id
 * @param {string} url - Gif's url
 * @param {string} title - Gif's title
 */
function Gif({id, url, title}) {
  return (
    <Link to={`/gif/${id}`} className="Gif">
      <h4>{title || 'No title ...'}</h4>
      <img src={url} alt={title} />
    </Link>
  );
}

Gif.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default memo(Gif);
