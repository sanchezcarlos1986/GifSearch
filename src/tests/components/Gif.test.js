import React from 'react';
import Gif from '../../components/Gif';
import {shallow} from 'enzyme';

describe('Gif', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      id: '123',
      title: 'Mega Man',
      url: 'http://myurl.com',
    };
    component = shallow(<Gif {...props} />);
  });

  it('should renders ok', () => {
    expect(component).not.toBeNull();
    expect(component.hasClass('Gif')).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it('should renders the image', () => {
    expect(component.find('img').exists()).toBe(true);
    expect(component.find('img').props().src).toBe(props.url);
    expect(component.find('img').props().alt).toBe(props.title);
  });

  it('should renders the title', () => {
    expect(component.find('h4').exists()).toBe(true);
    expect(component.find('h4').text()).toBe(props.title);
  });
});
