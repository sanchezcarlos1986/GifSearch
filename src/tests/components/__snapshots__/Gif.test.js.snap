// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif should renders ok 1`] = `ShallowWrapper {}`;
